File: frontend/lib/models/account.dart
class Account {
  final String id;
  final String bankName;
  final String accountType;
  final String branch;
  final DateTime openingDate;
  final String status;
  final bool linkedServices;

  Account({
    required this.id,
    required this.bankName,
    required this.accountType,
    required this.branch,
    required this.openingDate,
    required this.status,
    required this.linkedServices,
  });

  factory Account.fromJson(Map<String, dynamic> json) {
    return Account(
      id: json['_id'] ?? '',
      bankName: json['bankName'] ?? '',
      accountType: json['accountType'] ?? '',
      branch: json['branch'] ?? '',
      openingDate: DateTime.tryParse(json['openingDate'] ?? '') ?? DateTime.now(),
      status: json['status'] ?? 'Active',
      linkedServices: json['linkedServices'] ?? true,
    );
  }
}

File: backend/models/Account.js
const mongoose = require("mongoose");

const accountSchema = new mongoose.Schema(
  {
    customerId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Customer",
      required: true,
    },
    bankName: {
      type: String,
      required: true,
    },
    accountType: {
      type: String,
      enum: ["Savings", "Current"],
      required: true,
    },
    branch: {
      type: String,
      required: true,
    },
    openingDate: {
      type: Date,
      required: true,
    },
    status: {
      type: String,
      enum: ["Active", "Inactive"],
      default: "Active",
    },
    linkedServices: {
      type: Boolean,
      default: true,
    },
  },
  { timestamps: true }
);

module.exports = mongoose.model("Account", accountSchema);
